modules/percolator/src/main/java/org/elasticsearch/percolator/PercolatorHighlightSubFetchPhase.java

```diff
@@ -46,13 +47,10 @@
 
 // Highlighting in the case of the percolate query is a bit different, because the PercolateQuery itself doesn't get highlighted,
 // but the source of the PercolateQuery gets highlighted by each hit containing a query.
-public class PercolatorHighlightSubFetchPhase implements FetchSubPhase {
```

Comment:
+1 to extend from `HighlightPhase` instead of wrapping it in order to avoid injecting it!